### DISCUSSION TITLE
Where does the mod store the "Slots" to which you can save a backpack's "memory" tab configuration?

### DISCUSSION THREAD

**Discussion by Twisted-Code** [P3pp3rF1y/SophisticatedBackpacks#946] [Q&A]
![image](https://github.com/P3pp3rF1y/SophisticatedBackpacks/assets/7110033/f6dc65b2-1800-4235-97cd-9e8fd1f0c26c)
If it's somewhere client-side, I've been unable to find it. This is unfortunate, since I'd like to export it in case I want to use the same configuration on another server, or maybe even (if it's human readable, e.g. a .json file or something) edit it in Notepad++ or the like.

**Comment by P3pp3rF1y** [comment]
this is in backpack data `sophisticatedbackpacks.dat`, newer version adds to that saving to data packs as well (in newer version the saving is moved to saving the whole settings)

**Comment by Twisted-Code** [comment]
Okay I assume that's server-side then, and there is no way for me to export it normally (I don't see that file when I search my client directory. Even in `./saves/`, though to be fair I haven't used backpacks in single player.)? In my case, that's not a problem since I'm good friends with the server admin, but I'll probably follow this up with a suggestion to make user data like this accessible to the client.

**Reply by P3pp3rF1y** [reply]
I mean saving potentially could be done to client, but then loading can't because you need to load to server side state of the backpacks otherwise there would be need for reliance on unreliable client side data and loading would be much more complicated having to sync the data over network. So I am definitely not in favor of making this change as in addition to that I haven't really seen good arguments for why to export to client when in most of the cases one can either get it from server or recreate in single player and get the save there.

**Reply by Twisted-Code** [reply]
Regarding your argument about saving and loading (much of which I was already kind of anticipating), you may want to take a look at how Hexal, the Hexcasting add-on, handles the Everbook data, a cross-world file full of personal spells/macros. I don't know if it's perfectly analogous to your situation, but it seems pretty darn close. I know what you mean about "unreliable client data", but it would it be fair to say you only need to validate, not completely trust that client data, since it's not like manipulation would allow the creation of items?
On the other hand, your argument about other ways we can already get the data, current workarounds, is fair. I suppose if I run into a case where the server owner doesn't want to give me frankly not-at-all-sensitive data (for whatever strange reason), I can resort to single player as you said. Still, can we call export-to-client  (and preferably including the inverse operation, as complicated as the validation could be) a "nice to have"? 
anyway, for now at least I have the data I wanted, which was basically a backpack memory config that includes all possible gems from apotheosis, so I'm happy. If nothing else I can at least transcribe that manually onto another server, right?
